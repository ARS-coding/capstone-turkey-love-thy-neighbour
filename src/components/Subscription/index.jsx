import React from "react";

import { Container, Row, Col } from "react-bootstrap";

import { ReactComponent as PhoneAndNotificationsImg } from "../../images/PhoneandNotifications.svg";
import { ReactComponent as CheckIcon } from "../../images/Check.svg";

import "./index.scss";

function Subscription() {
  const [email, setEmail] = React.useState("");

  const handleChange = (event) => {
    setEmail(event.target.value);
  };
  return (
    <section className="subscription-img-bg container-fluid">
      <Container className="subscription-content-container d-flex align-items-center">
        <Row className="subscription-content-row m-0">
          <Col
            xs={12}
            sm={12}
            md={6}
            className="first-column d-flex flex-column justify-content-center align-items-start"
          >
            <h1>
              Subscribe to our <br /> newsletter.
            </h1>
            <h3>Never miss an update!</h3>
            <p>Signup to our newsletter to get a weekly digest.</p>
            <div className="button-input-wrapper">
              <input
                type="email"
                className="form-email"
                aria-describedby="emailHelp"
                placeholder="Email"
                value={email}
                onChange={handleChange}
              />
              <button
                type="submit"
                className="btn btn-primary"
                disabled={!email}
              >
                <CheckIcon className="check-icon" />
              </button>
            </div>
          </Col>

          <Col
            xs={12}
            sm={12}
            md={6}
            className="my-md-auto d-flex d-sm-flex d-md-block justify-content-center justify-content-sm-center justify-content-md-center"
          >
            <PhoneAndNotificationsImg />
          </Col>
        </Row>
      </Container>
    </section>
  );
}

export default Subscription;
