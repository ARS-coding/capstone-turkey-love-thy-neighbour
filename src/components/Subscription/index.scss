@import "../../style/variables";

//laptop background
@media screen and (min-width: 992px) {
  .subscription-img-bg {
    background-image: url("../../images/laptop-background-image.png");
    background-size: cover;
    background-size: contain;
    height: calc(100vh - 120px) !important;
    background-repeat: no-repeat;
  }
}

// tablet&phone background
.subscription-img-bg {
  @media screen and (min-width: 427px) and (max-width: 767px) {
    background-image: url("../../images/PhoneandNotificationsBG.png");
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-position: center right;
    background-size: 35%;
    svg:not(.check-icon) {
      display: none;
    }
  }

  @media screen and (max-width: 426px) {
    background-image: url("../../images/PhoneandNotificationsBG.png");
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-position: top 40% right;
    background-size: 200px 250px;
    svg:not(.check-icon) {
      display: none;
    }
  }
}

.subscription-img-bg {
  height: calc(100vh - 120px);

  .subscription-content-container {
    height: 100%;

    .subscription-content-row {
      & > div {
        padding: 0;
      }
      h1 {
        color: $dark-blue;
        font-weight: bold;
        @media screen and (min-width: 1200px) {
          font-size: 50px;
          margin-bottom: 115px;
        }
        @media screen and (min-width: 992px) and (max-width: 1199px) {
          font-size: 48px;
          margin-bottom: 115px;
        }
        @media screen and (min-width: 427px) and (max-width: 991px) {
          font-size: 38px;
          padding-bottom: 220px;
          padding-top: 50px;
        }
        @media screen and (min-width: 376px) and (max-width: 426px) {
          font-size: 34px;
          padding-bottom: 270px;
        }
        @media screen and (max-width: 375px) {
          font-size: 30px;
          padding-bottom: 270px;
        }
      }

      h3 {
        color: $logo-red;
        font-weight: bold;
        @media screen and (min-width: 991px) {
          font-size: 25px;
        }
      }

      p {
        color: $black;
        @media screen and (max-width: 991px) {
          font-size: 14px;
        }
      }
      @media screen and (max-width: 375px) {
        font-size: 30px;
        padding-bottom: 380px;
      }
    }

    svg:not(.check-icon) {
      @media screen and (min-width: 768px) {
        width: 100%;
        height: 450px !important;
      }
    }
  }
}
