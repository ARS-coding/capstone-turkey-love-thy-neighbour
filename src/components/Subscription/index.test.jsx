import React from "react";
import renderer from "react-test-renderer";
import Subscription from ".";

it("renders correctly", () => {
  const tree = renderer.create(<Subscription />).toJSON();
  expect(tree).toMatchSnapshot();
});
