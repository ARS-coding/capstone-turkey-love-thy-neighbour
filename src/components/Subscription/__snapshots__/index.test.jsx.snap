// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription Section renders Subscription section correctly 1`] = `
<section
  className="subscription-img-bg container-fluid"
>
  <div
    className="subscription-content-container d-flex align-items-center container"
  >
    <div
      className="subscription-content-row m-0 row"
    >
      <div
        className="first-column d-flex flex-column justify-content-center align-items-start col-md-6 col-sm-12 col-12"
      >
        <h1>
          Subscribe to our 
          <br />
           newsletter.
        </h1>
        <h3>
          Never miss an update !
        </h3>
        <p>
          Signup to our newsletter to get a weekly digest.
        </p>
        <div
          className="button-input-wrapper"
        >
          <input
            aria-describedby="emailHelp"
            className="form-email"
            onChange={[Function]}
            placeholder="E-Mail"
            type="email"
            value=""
          />
          <button
            className="btn btn-primary"
            disabled={true}
            type="submit"
          >
            <svg>
              Check.svg
            </svg>
          </button>
        </div>
      </div>
      <div
        className="my-md-auto d-flex d-sm-flex d-md-block justify-content-center justify-content-sm-center justify-content-md-center col-md-6 col-sm-12 col-12"
      >
        <img
          alt="phoneimg"
          className="img-fluid"
          src="PhoneandNotifications.svg"
        />
      </div>
    </div>
  </div>
</section>
`;
